cinema/tests/test_movie_and_movie_session.py

```diff
@@ -0,0 +1,286 @@
+import datetime
+
+from django.db.models import Count, F
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.test import APIClient
+from rest_framework.reverse import reverse
+
+from cinema.models import Movie, MovieSession, Actor, Genre, CinemaHall
+from cinema.serializers import MovieListSerializer, MovieSessionListSerializer
+from user.models import User
+
+MOVIE_LIST_URL = reverse("cinema:movie-list")
+MOVIE_SESSION_LIST_URL = reverse("cinema:moviesession-list")
+
+
+class MovieAndMovieSessionTestCaseForAnonimUser(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_movie_if_anonim_user(self):
+        movie_response = self.client.get(MOVIE_LIST_URL)
+
+        self.assertEqual(movie_response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_movie_sessions_if_anonim_user(self):
+        movie_sessions_response = self.client.get(MOVIE_SESSION_LIST_URL)
+
+        self.assertEqual(
+            movie_sessions_response.status_code, status.HTTP_401_UNAUTHORIZED
+        )
+
+
+class MovieAndMovieSessionTestCaseForSimpleUser(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = User.objects.create_user(
+            email="<EMAIL>",
+            password="<PASSWORD>",
+        )
```

Comment:
Hardcoding the user credentials here may not be a good practice. Consider using environment variables or a config file to store such sensitive information.